lib/server-args.js

```diff
@@ -0,0 +1,348 @@
+import fs from 'fs';
+import _ from 'lodash';
+import { DEFAULT_BASE_PATH } from 'appium-base-driver';
+
+// serverArgs will be added to the `server` (default) subcommand
+function parseSecurityFeatures (features) {
+  const splitter = (splitOn, str) => `${str}`.split(splitOn).map((s) => s.trim()).filter(Boolean);
+  let parsedFeatures;
+  try {
+    parsedFeatures = splitter(',', features);
+  } catch (err) {
+    throw new Error('Could not parse value of --allow/deny-insecure. Should be ' +
+                    'a list of strings separated by commas, or a path to a file ' +
+                    'listing one feature name per line.');
+  }
+
+  if (parsedFeatures.length === 1 && fs.existsSync(parsedFeatures[0])) {
+    // we might have a file which is a list of features
+    try {
+      const fileFeatures = fs.readFileSync(parsedFeatures[0], 'utf8');
+      parsedFeatures = splitter('\n', fileFeatures);
+    } catch (err) {
+      throw new Error(`Attempted to read --allow/deny-insecure feature names ` +
+                      `from file ${parsedFeatures[0]} but got error: ${err.message}`);
+    }
+  }
+
+  return parsedFeatures;
+}
+
+function parseDefaultCaps (caps) {
+  try {
+    // use synchronous file access, as `argparse` provides no way of either
```

Comment:
it is synchronous only in scope of the current process, but not cross-process

Reply:
this code was just moved over, it is not new. i'd prefer not to think about whether this needs re-architecting as part of this PR.